sdk/dapp-kit/src/hooks/wallet/useConnectWallet.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { UseMutationOptions } from '@tanstack/react-query';
+import { useMutation } from '@tanstack/react-query';
+import type {
+	StandardConnectInput,
+	StandardConnectOutput,
+	WalletAccount,
+	WalletWithSuiFeatures,
+} from '@mysten/wallet-standard';
+import { useWalletContext } from '../../components/WalletProvider.js';
+import { WalletAlreadyConnectedError } from '../../errors/walletErrors.js';
+import { setMostRecentWalletConnectionInfo } from '../../utils/walletUtils.js';
+import { walletMutationKeys } from '../../constants/walletMutationKeys.js';
+
+type ConnectWalletArgs = {
+	/** The wallet to connect to. */
+	wallet: WalletWithSuiFeatures;
```

Comment:
I think I might need to do a bit of a broader refactor since this doesn't technically have to be a wallet with Sui features, but I can do this afterwards

Reply:
Edit: Handled in https://github.com/MystenLabs/sui/pull/13721